crates/indexer/src/geyser/instructions/buy.rs

```diff
@@ -54,6 +54,34 @@ pub(crate) async fn process(
     };
 
     let values = row.clone();
+    let purchase_id;
+
+    let purchase_ids = client
+        .db()
+        .run({
+            let row = row.clone();
+            move |db| {
+                purchases::table
+                    .filter(
+                        purchases::buyer
+                            .eq(row.wallet.clone())
+                            .and(purchases::auction_house.eq(row.auction_house))
+                            .and(purchases::metadata.eq(row.metadata))
+                            .and(purchases::price.eq(row.buyer_price))
+                            .and(purchases::token_size.eq(row.token_size)),
+                    )
+                    .select(purchases::id)
+                    .load::<Uuid>(db)
+                    .context("failed to get purchase ids")
+            }
+        })
+        .await?;
```

Comment:
```suggestion
    let purchase_id = client
        .db()
        .run({
            let row = row.clone();
            move |db| {
                purchases::table
                    .filter(
                        purchases::buyer
                            .eq(row.wallet.clone())
                            .and(purchases::auction_house.eq(row.auction_house))
                            .and(purchases::metadata.eq(row.metadata))
                            .and(purchases::price.eq(row.buyer_price))
                            .and(purchases::token_size.eq(row.token_size)),
                    )
                    .select(purchases::id)
                    .first::<Uuid>(db)
                    .optional()
                    .context("failed to get purchase ids")
            }
        })
        .await?;
```

Will return the first match or none if not found. That should take care of the lower checks and assignment.

Reply:
Oh its working while .first::<Option<Uuid>> was giving error